app/models/application_choice.rb

```diff
@@ -169,6 +169,15 @@ def unconditional_offer?
     offer&.fetch('conditions', []).blank?
   end
 
+  # To be removed after testing. This has been tested here
+  # remove_application_choices_in_the_incorrect_cycle_spec.rb
+
+  def self.return_application_choices_with_a_course_in_a_different_cycle
+    ApplicationChoice
+    .joins(:application_form, course_option: [:course])
```

Comment:
So in the previous PR you mentioned using current_course_option.

I ran a query and there are two. I'd like to investigate those separately as they really shouldn't have been able to get into that state through the change offer flow.

I've updated the invariant check to use `current_course_option ` instead though!